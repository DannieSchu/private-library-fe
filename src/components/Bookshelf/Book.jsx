import React from 'react';
import PropTypes from 'prop-types';

const Book = ({ title, author, genre, pages }) => (
  <section>
    <h3>{title}</h3>
    <p>Author: {author}</p>
    <p>Genre: {genre}</p>
    <p>Number of pages: {pages}</p>
  </section>
);

Book.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  genre: PropTypes.string.isRequired,
  pages: PropTypes.number.isRequired,
};

export default Book;

