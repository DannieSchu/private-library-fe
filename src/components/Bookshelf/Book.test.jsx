import React from 'react';
import { shallow } from 'enzyme';
import Book from './Book';

describe('Book component', () => {
  it('renders Book', () => {
    const wrapper = shallow(<Book title="The Idiot" genre="Literature" pages={450} />);
    expect(wrapper).toMatchSnapshot();
  });
});
