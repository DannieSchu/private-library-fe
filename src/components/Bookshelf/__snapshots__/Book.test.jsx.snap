// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book component renders Book 1`] = `
<section>
  <h3>
    The Idiot
  </h3>
  <p>
    Author: 
  </p>
  <p>
    Genre: 
    Literature
  </p>
  <p>
    Number of pages: 
    450
  </p>
</section>
`;
